torch/csrc/jit/script/ir_emitter.cpp

```diff
@@ -1871,11 +1881,37 @@ struct to_ir {
           {rhs},
           {},
           self);
-
+    } else if (lhsValue->type()->kind() == TypeKind::ClassType) {
+      // Call `__iadd__` so updates happen in place on class types
+      // https://docs.python.org/3/reference/datamodel.html#object.__iadd__
+      auto magic_method_name = getAugMagicMethod(stmt);
+      const auto rhs = emitExpr(stmt.rhs());
+
+      // Insert call to magic method
+      auto sugared_magic_method = makeMagic(
```

Comment:
I think this also needs to handle classes which define `__add__`.  I added a test case above. The pattern here is:

 Does class define `__iadd__` ?

yes: call it 
no: does class define `__add__` ?

yes: call it
no: error